### PR TITLE
mimic: osd: take heartbeat_lock when calling heartbeat()

### PR DESCRIPTION
**Pull request by pdvian**
http://tracker.ceph.com/issues/39513

### CODE REVIEW

File: src/osd/OSD.cc [hunk 1]
```diff
@@ -5312,7 +5313,10 @@ void OSD::_preboot(epoch_t oldest, epoch_t newest)
 	   << oldest << ".." << newest << dendl;
 
   // ensure our local fullness awareness is accurate
-  heartbeat();
+  {
+    Mutex::Locker l(heartbeat_lock);
```
Comment (pdvian) [inline src/osd/OSD.cc]: @liewegas need review. std::lock_guard is not being recognized in mimic.

Comment (smithfarm) [reply src/osd/OSD.cc]: @pdvian In addition to "resolved for <function name>" in the Conflicts section, could you also say something like "std::lock_guard is not being recognized in mimic; used Mutex::Locker instead"?

**Comment by yuriw** [comment]
[wip-yuri-testing-2019-06-26-1612-mimic](https://trello.com/c/JaOACLNP/655-wip-yuri-testing-2019-06-26-1612-mimic)

**Comment by yuriw** [comment]
@pdvian this passed tests and can be merged as soon as @smithfarm comment is resolved

**Comment by yuriw** [comment]
[wip-yuri-testing-2019-07-08-2007-mimic](https://trello.com/c/QAzBp49o/659-wip-yuri-testing-2019-07-08-2007-mimic)

**Pull request merged**